src/ontodia/diagram/defaultLinkRouter.ts

```diff
@@ -0,0 +1,218 @@
+import * as joint from 'jointjs';
+import { LinkRouter, Vertex, RouterProps } from '../customization/props';
+import { DiagramModel } from './model';
+import { Link, Element } from './elements';
+import { LinkView } from './linkView';
+
+const LEFT_LABEL_POSITION = {
```

Comment:
It would be safer to declare functions returning new literals each time instead. Otherwise we could accidentally overwrite   properties of this constants by deep merging link styles.